File: WindowsGrep/WindowsGrep.Core/Enums/NativeCommandType.cs
﻿namespace WindowsGrep.Core;

public enum NativeCommandType
{
    [DescriptionCollection("ls")]
    List,

    [ExpectsParameter(true)]
    [DescriptionCollection("cd")]
    ChangeDirectory,

    [DescriptionCollection("clear")]
    ClearConsole,

    [DescriptionCollection("pwd")]
    PrintWorkingDirectory
}

File: WindowsGrep/src/enums/native_command_type.rs
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum NativeCommandType {
    List,
    ChangeDirectory,
    ClearConsole,
    PrintWorkingDirectory,
}

pub struct NativeCommandInfo {
    pub command_type: NativeCommandType,
    pub descriptor: &'static str,
    pub expects_parameter: bool,
}

pub const NATIVE_COMMANDS: &[NativeCommandInfo] = &[
    NativeCommandInfo {
        command_type: NativeCommandType::List,
        descriptor: "ls",
        expects_parameter: false,
    },
    NativeCommandInfo {
        command_type: NativeCommandType::ChangeDirectory,
        descriptor: "cd",
        expects_parameter: true,
    },
    NativeCommandInfo {
        command_type: NativeCommandType::ClearConsole,
        descriptor: "clear",
        expects_parameter: false,
    },
    NativeCommandInfo {
        command_type: NativeCommandType::PrintWorkingDirectory,
        descriptor: "pwd",
        expects_parameter: false,
    },
];
